google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -1112,30 +1228,36 @@ def gapi_from_attrs *attributes
           Google::Apis::StorageV1::Object.new attr_params
         end
 
-        def rewrite_gapi bucket, name, update_gapi
+        def rewrite_gapi bucket, name, updated_gapi,
+                         new_bucket: nil, new_name: nil, acl: nil,
+                         generation: nil, encryption_key: nil,
+                         new_encryption_key: nil, user_project: nil
+          new_bucket ||= bucket
+          new_name ||= name
+          options = { acl: File::Acl.predefined_rule_for(acl),
+                      generation: generation, source_key: encryption_key,
+                      destination_key: new_encryption_key,
+                      user_project: user_project }.delete_if { |_k, v| v.nil? }
+
           resp = service.rewrite_file \
-            bucket, name, bucket, name, update_gapi, user_project: user_project
+            bucket, name, new_bucket, new_name, updated_gapi, options
           until resp.done
             sleep 1
+            retry_options = options.merge token: resp.rewrite_token
             resp = service.rewrite_file \
-              bucket, name, bucket, name, update_gapi,
-              token: resp.rewrite_token, user_project: user_project
+              bucket, name, new_bucket, new_name, updated_gapi, retry_options
           end
           resp.resource
         end
 
-        def fix_copy_args dest_bucket, dest_path, options = {}
-          if dest_path.respond_to?(:to_hash) && options.empty?
```

Comment:
What was the story with the `options` param and return value, and the `dest_path` here? I'm assuming this is no longer needed, but what was it for?